ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoLogger.java

```diff
@@ -30,58 +32,47 @@
  * @author Joris Mancini {@literal <joris.mancini at rte-france.com>}
  */
 final class SearchTreeRaoLogger {
-
     private SearchTreeRaoLogger() {
     }
 
-    static void logRangeActions(Leaf leaf, Set<RangeAction> rangeActions) {
-        logRangeActions(leaf, rangeActions, null);
+    static void logRangeActions(FaraoLogger logger,
+                                Leaf leaf, Set<RangeAction> rangeActions) {
+        logRangeActions(logger, leaf, rangeActions, null);
     }
 
-    static void logRangeActions(Leaf leaf, Set<RangeAction> rangeActions, String prefix) {
-        StringBuilder rangeActionMsg = new StringBuilder();
-        if (prefix != null) {
-            rangeActionMsg.append(prefix).append(" - ");
-        }
-        rangeActionMsg.append("Range action(s): ");
-        rangeActions.forEach(rangeAction -> {
-            String rangeActionName = rangeAction.getName();
+    static void logRangeActions(FaraoLogger logger, Leaf leaf, Set<RangeAction> rangeActions, String prefix) {
+        String rangeActionSetpoints = rangeActions.stream().map(rangeAction -> {
             if (rangeAction instanceof PstRangeAction) {
                 int rangeActionTap = leaf.getOptimizedTap((PstRangeAction) rangeAction);
-                rangeActionMsg
-                        .append(format("%s: %d", rangeActionName, rangeActionTap))
-                        .append(" , ");
+                return format("%s: %d", rangeAction.getName(), rangeActionTap);
             } else {
                 double rangeActionSetPoint = leaf.getOptimizedSetPoint(rangeAction);
-                rangeActionMsg
-                        .append(format("%s: %.2f", rangeActionName, rangeActionSetPoint))
-                        .append(" , ");
-
+                return format(Locale.ENGLISH, "%s: %.2f", rangeAction.getName(), rangeActionSetPoint);
             }
-        });
-        String rangeActionsLog = rangeActionMsg.toString();
-        SearchTree.LOGGER.info(rangeActionsLog);
+        }).collect(Collectors.joining(", "));
+        logger.info("{}range action(s): {}", prefix == null ? "" : prefix, rangeActionSetpoints);
```

Comment:
should a space be added between the prefix and "range action(s)"?

Reply:
i chose to let the user include the space in the prefix; it would be better if he doesn't want a prefix at all

Reply:
good point, I hadn't thought of the case with no prefix